improver/utilities/copy_attributes.py

```diff
@@ -0,0 +1,55 @@
+# (C) Crown copyright, Met Office. All rights reserved.
+#
+# This file is part of improver and is released under a BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+from typing import List, Tuple, Union
+
+from iris.cube import Cube, CubeList
+
+from improver import BasePlugin
+from improver.metadata.amend import amend_attributes
+from improver.utilities.common_input_handle import as_cube, as_cubelist
+
+
+class CopyAttributes(BasePlugin):
+    """Copy attribute values from template_cube to cube, overwriting any existing values."""
+
+    def __init__(self, attributes: List):
+        """
+        Initialise the plugin with a list of attributes to copy.
+
+        Args:
+            attributes:
+                List of names of attributes to copy. If any are not present on template_cube, a
+                KeyError will be raised.
+        """
+        self.attributes = attributes
+
+    def process(
+        self, *cubes: Union[Cube, CubeList], template_cube: Union[Cube, CubeList]
+    ) -> Union[Tuple[Union[Cube, CubeList]], Cube, CubeList]:
```

Comment:
Notice how for plugins that can accept any number of inputs, I would propose this return behaviour.
That being, return a Cube or CubeList, if only 1 input was provided, otherwise, return a tuple of Cube/CubeList objects corresponding to our inputs arguments.